workshop/Lab4/osbapi.md

```diff
@@ -0,0 +1,68 @@
+# Open Service Broker API
+
+Cloud native applications are ephemeral but still require the use of stable storage, databases,
+caches, email systems, billing systems, etc. Developers need to provide for these needs somehow,
+but it would be preferable if they could focus on the thing they actually care about: developing
+their application. Similarly, Cloud platform providers need some way to provide for these needs,
+but are more rightly more concerned with managing their platform.
+
+The Open Service Broker aims to solve these problems by providing a open standard to create brokers
+that allow application developers to provision, maintain, and consume services in a without having 
+to know what's going on behind the scenes. If you need a Redis store, you can request one from
+a service broker, bind it to your application, and go. Anything that can be decomposed into the
+service broker action lifecycle of provison, bind, unbdind, and deprovision can be offered as an
+OSB service, from databases to subscriptions to web APIs.
+
+OSB API describes five object types: brokers, services, plans, instances, and bindings.
+
+* Brokers are a server that implements the OSB API and offers available services, e.g. a Redis broker.
+* Services are a category of services offered by a broker, e.g. Redis stores. In Kubernetes,
+these are referred to as Service Classes.
+* Plans are a specific type of a Service that a Broker offers, e.g. version 4.0.9 of Redis.
```

Comment:
Plans allow a way for a Service to offer a set of different variants of the Service - typically each Plan offers a different quality of service and each might have a different price associated with it.

Reply:
I would use versions of Redis as an example even though its possible to think of a Plan that way, I think its more likely to be a QoS type of thing

Reply:
what would you use as an example then?

Reply:
Well, you don't have to mention price as the key differentiation - rather its the QoS, which could also imply a different price.  Just to make it real.

w.r.t example/text - just use what you say at the start of the labs... they're usually used for QoS differentiation.